src/read/wasm.rs

```diff
@@ -54,23 +63,229 @@ impl<'data> WasmFile<'data> {
+                                        export.index.checked_sub(imported_funcs_count)
+                                    {
+                                        let local_func_kind =
+                                            &mut local_func_kinds[local_func_id as usize];
+                                        if let LocalFunctionKind::Unknown = local_func_kind {
+                                            *local_func_kind = LocalFunctionKind::Exported {
+                                                symbol_ids: Vec::new(),
+                                            };
+                                        }
+                                        let symbol_ids = match local_func_kind {
+                                            LocalFunctionKind::Exported { symbol_ids } => {
+                                                symbol_ids
+                                            }
+                                            _ => unreachable!(),
+                                        };
+                                        symbol_ids.push(file.symbols.len() as u32);
+                                    }
+                                    SymbolKind::Text
+                                }
+                                wp::ExternalKind::Table
+                                | wp::ExternalKind::Memory
+                                | wp::ExternalKind::Global => SymbolKind::Data,
+                            },
+                            section: SymbolSection::Section(SectionIndex(SECTION_EXPORT)),
+                            weak: false,
+                            scope: SymbolScope::Dynamic,
+                            flags: SymbolFlags::None,
+                        });
+                    }
+                }
+                wp::SectionCode::Start => {
+                    entry_func_id = Some(
+                        section
+                            .get_start_section_content()
+                            .map_err(|_| "Couldn't read contents of the start section")?,
+                    );
+                }
+                wp::SectionCode::Code => {
+                    if let Some(main_file_symbol) = main_file_symbol.take() {
+                        file.symbols.push(main_file_symbol);
+                    }
+
+                    for (i, (body, local_func_kind)) in section
+                        .get_code_section_reader()
+                        .map_err(|_| "Couldn't read header of the code section")?
+                        .into_iter()
+                        .zip(&mut local_func_kinds)
+                        .enumerate()
+                    {
+                        let body = body.map_err(|_| "Couldn't read a function body")?;
+                        let range = body.range();
+
+                        let section = SymbolSection::Section(SectionIndex(SECTION_CODE));
+                        let address = range.start as u64;
```

Comment:
This appears to be using the file offset as the address? If so, then sections should do the same thing for their address, instead of all being 0.

Reply:
No, `range.start` at this point is relative to the code section (please check the example dumps in the PR).

Reply:
From the dump:
```
10: Section { name: "<code>", address: 0, size: 140, kind: Unknown }
...
9: Symbol { name: Some("h"), address: 220, size: 30, kind: Text, section: Section(SectionIndex(10)), weak: false, scope: Dynamic, flags: None }
```
220 can't be an offset from the code section because it is larger than the code section.

Reply:
Huh, good catch! I can swear it used to be relative to a code section, but either I changed a code or maybe that was a bug in some older version of wasmparser :/

I'll update the PR to use section-relative addresses later.

Reply:
> If so, then sections should do the same thing for their address, instead of all being 0.

Coming back to this - do we have any separate field for "file offset" as opposed to memory address? I don't know if Wasm is alone in this, but these are not the same since many sections don't actually get loaded into [observable / addressable] memory.

Reply:
`file_range` gives the file offset and size (which can be different from `size`).

Reply:
I've looked into this a bit more, and, since browsers use absolute addresses for stacktraces, I suppose that's what I will go with. DWARF is the odd one out, but if someone needs to calculate a relative address, they can do so from absolute ones.

Reply:
I'm still bothered by addresses vs file offsets though. Unified interfaces are hard 😩

Reply:
> do we have any separate field for "file offset" as opposed to memory address

Right, I forgot there is `file_offset`, it's just not printed out in `objdump` example.

Reply:
I went for code-section-relative addresses after all; if someone wants an absolute file offset, they can get it from `file_offset`, but if they want an address of the function within code space (e.g. for DWARF), they can use `address`. I think this is closest we can get to in terms of what other object files provide.